.github/workflows/towncrier-changelog.yml

```diff
@@ -0,0 +1,46 @@
+name: Towncrier changelog
+
+on:
+  workflow_call:
+    inputs:
+        app_version:
+            type: string
+            description: 'The version of the app'
+            required: true
+        repo:
+            type: string
+            description: 'The repository to update'
+            required: true
+        branch:
+            type: string
+            description: 'The branch to update'
+            required: true
+
+jobs:
+  create-changelog:
+    name: Create changelog with towncrier
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout ${{ inputs.repo }}
+        uses: actions/checkout@v4
+        with:
+          repository: substra/${{ inputs.repo }}
+          path: ${{ inputs.repo }}
```

Comment:
Not sure what this line is doing, is it checking out a subset?

Reply:
It clone the repu into a folder `path`. I think it is useful when we clone several repo but in this case you're right I think we can remove it and remove the `cd ${{ inputs.repo }}` in the `Run Towncrier` step.
I'll do the changes

Reply:
Or maybe it isolating the repo in a folder is a good thing ? Maybe we can keep it to ensure that...

Reply:
I checked several implementation and we always define the `path`. I think it's a good practice to keep